irobot_events_executor/src/rclcpp/timers_manager.cpp

```diff
@@ -61,11 +61,7 @@ void TimersManager::stop()
 {
   // Lock stop() function to prevent race condition in destructor
   std::unique_lock<std::mutex> lock(stop_mutex_);
-  // Nothing to do if the timers thread is not running
-  // or if another thread already signaled to stop.
-  if (!running_.exchange(false)) {
-    return;
-  }
+  running_ = false;
```

Comment:
The reason for doing `running_.exchange(false)` is that I wanted to avoid to send multiple signals to stop the threads.

It's not for performance reasons, but rather because two threads shouldn't be allowed to stop the timers manager at the same time.
However, I do see the problem:
 - thread 1 stops the timers manager
 - thread 2 stops the timers manager while this is already stopping
 - thread 2 returns sooner than the timers manager has already stopped and does something bad assuming it has already fully stopped.